doc/conf.py

```diff
@@ -165,6 +183,33 @@
 # Output file base name for HTML help builder.
 htmlhelp_basename = 'theanodoc'
 
+# Options for the linkcode extension
+# ----------------------------------
+# Resolve function
+# This function is used to populate the (source) links in the API
+def linkcode_resolve(domain, info):
+    def find_source():
+        # try to find the file and line number, based on code from numpy:
+        # https://github.com/numpy/numpy/blob/master/doc/source/conf.py#L286
+        obj = sys.modules[info['module']]
+        for part in info['fullname'].split('.'):
+            obj = getattr(obj, part)
+        import inspect
+        import os
+        fn = inspect.getsourcefile(obj)
+        fn = os.path.relpath(fn, start=os.path.dirname(theano.__file__))
+        source, lineno = inspect.getsourcelines(obj)
+        return fn, lineno, lineno + len(source) - 1
+
+    if domain != 'py' or not info['module']:
+        return None
+    try:
+        filename = 'theano/%s#L%d-L%d' % find_source()
+    except Exception:
+        filename = info['module'].replace('.', '/') + '.py'
+    from subprocess import check_output
+    tag = check_output('git rev-parse HEAD', shell=True)[:-1]
+    return "https://github.com/Theano/theano/blob/%s/%s" % (tag, filename)
```

Comment:
This does not work correctly with Python 3, I have a link that looks like `https://github.com/Theano/theano/blob/b'b6a7bd28fe014655b358a66f20ae535821dfb439'/theano/compile/function.py#L76-L326` because the commit number gets printed as `b'b6a7bd28fe014655b358a66f20ae535821dfb439'`